sqlmesh/core/renderer.py

```diff
@@ -445,35 +456,31 @@ def render(
             runtime_stage=runtime_stage,
             **kwargs,
         )
-
-        if not isinstance(query, exp.Subqueryable):
-            raise_config_error(f"Query needs to be a SELECT or a UNION {query}.", self._path)
-
         return query
 
     def update_cache(
         self,
-        expression: exp.Expression,
+        expression: t.Optional[exp.Expression],
         start: t.Optional[TimeLike] = None,
         end: t.Optional[TimeLike] = None,
         execution_time: t.Optional[TimeLike] = None,
         optimized: bool = False,
         **kwargs: t.Any,
     ) -> None:
-        if not optimized:
+        if optimized:
+            if not isinstance(expression, exp.Subqueryable):
```

Comment:
Can this ever happen? Or just a precaution?

Reply:
internal error + types